### PR TITLE
Remove unused AsEquivalent* conversion methods

### PR DESCRIPTION
**Pull request by fingolfin**
Closes #629

### CODE REVIEW

File: src/poly/poly.jl [hunk 1]
```diff
@@ -1171,29 +1171,6 @@ end
...
-Return an AbstractAlgebra (multivariate) polynomial ring over the base ring of $R$ in variables having the same names as those of R.
-"""
-function AsEquivalentAbstractAlgebraPolynomialRing(R::Singular.PolyRing{T}; ordering::Symbol = :degrevlex) where T <: Singular.n_unknown
-   return AbstractAlgebra.Generic.polynomial_ring(base_ring(R).base_ring,
-	       symbols(R), ordering=ordering)
-end
-
-
 @doc Markdown.doc"""
     (R::PolyRing){T <: RingElem}(p::AbstractAlgebra.Generic.MPoly{T})
```
Comment (fingolfin) [inline src/poly/poly.jl:1175]: I guess this code should also be removed

**Review by thofma — APPROVED** [review]
Breaking change? Not sure if there is code is out there using this (outside OSCAR land)

**Comment by fingolfin** [comment]
According to https://juliahub.com/ui/Search?q=AsEquivalent&type=code no other registered package is using this. But of course someone might have private code using it... ?

**Comment by fingolfin** [comment]
I don't see harm in marking this as breaking either, so, why not...

**Pull request merged**